lib/components/map/default-map.js

```diff
@@ -44,6 +46,64 @@ const MapContainer = styled.div`
     box-sizing: unset;
   }
 `
+/**
+ * Get the configured display names for the specified company ids.
+ */
+function getCompanyNames(companyIds, config, intl) {
+  return intl.formatList(
+    (companyIds || []).map(
+      (id) =>
+        config.companies?.find((company) => company.id === id)?.label || id
+    ),
+    { type: 'conjunction' }
+  )
+}
+
+/**
+ * Determines the localized name of a map layer by its type.
+ */
+function getLayerName(overlay, config, intl) {
+  // HACK: Support for street/satellite configs that use the name.
+  switch (overlay.name) {
+    case 'Streets':
+      return intl.formatMessage({ id: 'components.MapLayers.streets' })
+    case 'Satellite':
+      return intl.formatMessage({ id: 'components.MapLayers.satellite' })
+    default:
+    // No action in the default case.
+  }
+
+  switch (overlay.type) {
+    case 'streets':
+      return intl.formatMessage({ id: 'components.MapLayers.streets' })
+    case 'satellite':
+      return intl.formatMessage({ id: 'components.MapLayers.satellite' })
+    case 'bike-rental':
+      return intl.formatMessage(
+        { id: 'components.MapLayers.bike-rental' },
+        {
+          companies: getCompanyNames(overlay.companies, config, intl)
+        }
+      )
+    case 'car-rental':
+      return intl.formatMessage({ id: 'components.MapLayers.car-rental' })
+    case 'micromobility-rental':
+      return intl.formatMessage(
+        {
+          id: 'components.MapLayers.micromobility-rental'
+        },
+        {
+          companies: getCompanyNames(overlay.companies, config, intl)
+        }
+      )
+    case 'park-and-ride':
+      return intl.formatMessage({ id: 'components.MapLayers.park-and-ride' })
+    case 'stops':
+      return intl.formatMessage({ id: 'components.MapLayers.stops' })
+    default:
+      return null
```

Comment:
Do we really want the fallback to be null? Surely displaying the "raw" type is better than nothing at all?

Reply:
Good point. In https://github.com/opentripplanner/otp-react-redux/pull/606/commits/1162dc3bc6d0532c5ef6ad387f9c61d193527718, it defaults to `overlay.type`. (if `overlay.name` is specified for any overlay, that still preempts the default i18n messages.